lib/event_message.rb

```diff
@@ -5,12 +5,18 @@
 # Encapsulate a message from the work orders application to be sent to the mesage broker.
 class EventMessage
   attr_reader :work_order
+  attr_reader :catalogue
 
   ROUTING_KEY = 'aker.events.work_order'
 
   def initialize(params)
-    @work_order = params.fetch(:work_order)
-    @status = params.fetch(:status)
+    # For Work Order message
+    @work_order = params.fetch(:work_order, nil)
+    @status = params.fetch(:status, nil)
+
+    # For Catalogue message
+    @catalogue = params.fetch(:catalogue, nil)
+    @catalogue_error = params.fetch(:error, nil)
```

Comment:
This class really looks like it should be abstract (or at least not directly initialized as Ruby has not abstract classes). 

```ruby
class EventMessage

  def generate_json
     raise "This must be overridden!"
  end

end

class WorkOrderMessage < EventMessage

  def generate_json
    ...
  end

end

class CatalogueMessage < EventMessage

  def generate_json
    ...
  end

end
```

Or something like that. As more messages are added in the future having a big `if` statement depending on what params are set is not as maintainable as it could be.